lib/resque/worker_job_removal.rb

```diff
@@ -0,0 +1,113 @@
+module Resque
+  module WorkerJobRemoval
+    def self.prepended(base)
+      base.extend SmartIdMatcher
+    end
+
+    module SmartIdMatcher
+      attr_accessor :smart_id_matcher
+    end
+
+    def startup
+      super
+      start_kill_operations_thread if fork_per_job?
+    end
+
+    def perform_with_fork(job, &block)
+      @last_fork_time = Time.now.to_i
+      super
+    end
+
+    def work_one_job(job = nil, &block)
+      return false if paused?
+      return false unless job ||= reserve
+
+      if (name = suspend?(job))
+        Job.create("suspended:#{name}", job.payload['class'], *job.payload['args'])
```

Comment:
Indeed it would make sense to keep the suspended jobs in a failure queue instead of a normal one.

I made some tests with `FailedQueueAccess` in order to enable multiple failed queues and send the suspended jobs to a separate failed queue. The issue is that the failure backend class should change from the current `Resque::Failure::Redis` to `Resque::Failure::RedisMultiQueue` and this comes with multiple implications:

Failures from each normal queue are classified per queue. high_failed, medium_failed, low_failed etc.
Also, the failed overview classification view is broken and propably needs refactoring.

Should we invest more time on this? Wdyt?

Reply:
Aha, nice, no I don't think we should go down that road and use multi queue for failures.

I was thinking, very roughly, of adding to [DataStore](https://github.com/skroutz/resque/blob/skroutz/lib/resque/data_store.rb#L9) something like:
```
@suspended_queue_access = FailedQueueAccess.new
```
and then use that one to add/retrieve messages from the suspended list.

Though now that I think about it, it would require more changes to the DateStore and/or FailedQueueAccess classes, and it would diverge more from the upstream implementation, so I'm not sure that's worth it either. Wdyt?

Otherwise, if we keep the current implementation with `Job.create`, do you think that excluding such queues from the `queues` method covers our needs?

Reply:
> I was thinking, very roughly, of adding to [DataStore](https://github.com/skroutz/resque/blob/skroutz/lib/resque/data_store.rb#L9) something like:
> 
> @suspended_queue_access = FailedQueueAccess.new
> and then use that one to add/retrieve messages from the suspended list, specifying a different queue name.

We could do this or even use the existing `FailedQueueAccess` object to send the suspended jobs to a different failed queue but without changing the failures backend class to `Resque::Failure::RedisMultiQueue` the queue would not be visible to the resque interface.

> Otherwise, if we keep the current implementation with Job.create, do you think that excluding such queues from the queues method covers our needs?

I think that our needs are covered since the workers continue to run jobs from the rest normal queues.

Reply:
True. Ok I'd say let's proceed with the current implementation and see how it goes? 
We can always revisit this if we encounter any issues.